examples/vl3_basic/vl3-nse/cmd/service_registry.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/tiswanso/examples/api/serviceregistry"
+	"google.golang.org/grpc"
+	"strconv"
+	"strings"
+)
+
+const (
+	POD_NAME = "podName"
+	SERVICE_NAME = "service"
+	PORT = "port"
+	CLUSTER_NAME = "clusterName"
+)
+
+type validationErrors []error
+
+type ServiceRegistry interface {
```

Comment:
to be fair I don't see any reason to use this implementation, it's just a wrap around `RegistryClient`. Moreover these 2 methods have a lot of parameters (7 and with context 8). Generally it's not a good practice to have so may parameters

Reply:
reduced number of arguments to 4, including the context, but i kept the wrap